src/smallestPrimeFactor.js

```diff
@@ -11,15 +11,48 @@
  * @param {number} num - A number
  * @returns {number} The smallest prime factor of num
  */
-function smallestPrimeFactor(num) {
+
+  function isPrime(num) {
+    if ( num === 1){
+       return false;}
+
+       if ( num === 2){return true;}
+
+     for ( let n = 2; n < num; n++){
```

Comment:
You don't need to check all the way up to `num`. Consider something like `18`. Sure, `9` is a factor, but you'll never check for it because `18 / 9` is `2` and you would've checked for that earlier.

In general, if `k` is a factor of `num` then `num / k` is an integer and you have `num === (num / k) * k`. The larger `k` is the smaller `num / k` is. If `s` is the smallest proper factor of num, how big can `s` be?

If you haven't found a proper factor by that point you know you won't find a larger one (because its complement would be smaller and you would have already found it).